frontend/dockerfile/docs/reference.md

```diff
@@ -1160,41 +1160,96 @@ The available `[OPTIONS]` are:
 - [`--link`](#add---link)
 - [`--exclude`](#add---exclude)
 
-The `ADD` instruction copies new files, directories or remote file URLs from `<src>`
-and adds them to the filesystem of the image at the path `<dest>`.
+The `ADD` instruction copies new files or directories from `<src>` and adds
+them to the filesystem of the image at the path `<dest>`. Files and directories
+can be copied from the build context, a remote URL, or a Git repository.
 
-Multiple `<src>` resources may be specified but if they are files or
-directories, their paths are interpreted as relative to the source of
-the context of the build.
+> **Note**
+>
+> The first encountered `ADD` instruction will invalidate the cache for all
+> following instructions from the Dockerfile if the source has changed. This
```

Comment:
Invalidate cache for the following commands in the same stage is what you probably mean. But even that is not true for all cases, eg. `COPY --link`

Also, this behavior isn't really specific to `ADD` or `COPY` command.

Reply:
I can remove this, we cover cache invalidation elsewhere.